src/realm/object-store/c_api/realm.cpp

```diff
@@ -2,6 +2,13 @@
 #include "realm.hpp"
 
 
+#if !REALM_ENABLE_SYNC
+namespace realm {
+BindingCallbackThreadObserver* g_binding_callback_thread_observer = nullptr;
```

Comment:
There was a compilation error with the object store tests if REALM_ENABLE_SYNC was not enabled, since the binding_callback_thread_observer.cpp files were moved to _realm/src/sync_ and they are not imported in this case. This provides a dummy symbol so it can still compile.

Reply:
As an alternative, I could put the `#IF REALM_ENABLE_SYNC` condition around the binding callback thread observer C_API.

Reply:
Ah, hmm, this is because we moved the file into the sync directory? A quick look through github would suggest we don't actually use the binding callback thread observer from the C API - only from the C++ API, so maybe we can remove the C API functions altogether? I think it makes more sense to put the `#if REALM_ENABLE_SYNC` condition around the binding callback thread observer C API functions than to make a dummy duplicate symbol to fix a compile error.

Reply:
Removed the duplicate symbol and added `#if REALM_ENABLE_SYNC`'s